identity/src/keypair.rs

```diff
@@ -224,8 +236,13 @@ impl Keypair {
 
     /// Encode a private key as protobuf structure.
     pub fn to_protobuf_encoding(&self) -> Result<Vec<u8>, DecodingError> {
-        use quick_protobuf::MessageWrite;
+        self.encode_protobuf()
+    }
 
+    /// Encode a private key as protobuf structure.
+    ///
+    /// See <https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#key-types> for details on the encoding.
+    pub fn encode_protobuf(&self) -> Result<Vec<u8>, DecodingError> {
```

Comment:
It is a shame that we rename the function here and cannot do the signature change to infalliblity at the same time. For that we need to also implement RSA encoding but I think for that we need to switch to the `rsa` crate which I want to debate first.

I wonder:

- should we delay the rename of this function until then so we can land the rest now
- should we rename it with the others and send users through another rename (+ signature change) once we support RSA

Reply:
Delay the rename then.